### PR TITLE
feat(textarea): add md-autosize directive

### PR DESCRIPTION
**Pull request by jelbourn**
R: @mmalerba 

CC: @fxck

### CODE REVIEW

**Comment by fxck** [comment]
Yeah, LGTM.

File: src/lib/input/autosize.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    '[style.min-height]': '_minHeight',
+    '[style.max-height]': '_maxHeight',
+  },
+})
+export class MdTextareaAutosize implements OnInit {
+  /** Minimum number of rows for this textarea. */
+  @Input() minRows: number;
+
+  /** Maximum number of rows for this textarea. */
+  @Input() maxRows: number;
```
Comment (mmalerba) [inline src/lib/input/autosize.ts]: could you also do an @Input() for value so you can resize when the content is changed programmatically?

File: src/lib/input/autosize.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    textareaClone.rows = 1;
+
+    // Use `position: absolute` so that this doesn't cause a browser layout and use
+    // `visibility: hidden` so that nothing is rendered. Clear any other styles that
+    // would affect the height.
+    textareaClone.style.position = 'absolute';
+    textareaClone.style.visibility = 'hidden';
+    textareaClone.style.padding = '';
+    textareaClone.style.minHeight = '';
+    textareaClone.style.height = '';
```
Comment (mmalerba) [inline src/lib/input/autosize.ts]: also clear maxHeight and border

File: src/lib/input/autosize.ts [hunk 3]
```diff
@@ -19,10 +19,24 @@ export class MdTextareaAutosize implements OnInit {
   /** Maximum number of rows for this textarea. */
   @Input() maxRows: number;
 
+  /** The value of the textarea. */
+  private _value: string;
```
Comment (mmalerba) [inline src/lib/input/autosize.ts]: Is this our team's official style for these getter/setters? If it's still up for debate, I prefer:
```
/** ... */
@Input()
get value() {...}
set value(v) {...}
private _value;
```
It's a good reminder that I should be accessing it through the getter unless I have a good reason not to.

**Comment by mmalerba** [comment]
whoops, didn't notice this was still wip

**Comment by jelbourn** [comment]
@mmalerba as we discussed, adding support for changing on `value` change is problematic, going to leave it out for now.

@fxck `md-autosize` will simply override `rows` now, which I think is a sensible behavior.

**Comment by mmalerba** [comment]
lgtm

**Pull request merged**